titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeOpportunisticResourceProvider.java

```diff
@@ -92,22 +95,20 @@ public KubeOpportunisticResourceProvider(MesosConfiguration configuration, Titus
                 .scheduleOn(metricsPollerExecutor)
                 .monitorValue(this, KubeOpportunisticResourceProvider::currentOpportunisticCpuCount);
 
-        ApiClient apiClient = Config.fromUrl(configuration.getKubeApiServerUrl());
-        apiClient.getHttpClient().setReadTimeout(0, TimeUnit.SECONDS); // infinite timeout for watch calls
-        this.api = new CustomObjectsApi(apiClient);
+        ApiClient apiClient = createApiClient(configuration.getKubeApiServerUrl(), CLIENT_METRICS_PREFIX,
+                titusRuntime, refreshIntervalMs);
+        api = new CustomObjectsApi(apiClient);
 
-        AtomicLong nextThreadNum = new AtomicLong(0);
-        informerFactory = new SharedInformerFactory(apiClient, Executors.newCachedThreadPool(runnable -> {
-            Thread thread = new Thread(runnable, KubeOpportunisticResourceProvider.class.getSimpleName() + "-" + nextThreadNum.getAndIncrement());
-            thread.setDaemon(true);
-            return thread;
-        }));
+        informerFactory = createSharedInformerFactory(
+                KubeOpportunisticResourceProvider.class.getSimpleName() + "-",
+                apiClient
+        );
         // TODO(fabio): enhance the kube client to support custom JSON deserialization options
         informer = informerFactory.sharedIndexInformerFor(
                 this::listOpportunisticResourcesCall,
                 V1OpportunisticResource.class,
                 V1OpportunisticResourceList.class,
-                configuration.getKubeOpportunisticRefreshIntervalMs()
+                refreshIntervalMs
```

Comment:
should probably use a slightly larger interval here, just to avoid killing the connection in the middle of it trying to reconcile

Reply:
changed as it was being overriden